packages/graph/hash_graph/lib/graph/src/types/schema/array.rs

```diff
@@ -0,0 +1,313 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+enum ArrayTypeTag {
+    Array,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+struct Array {
```

Comment:
Perhaps we should call this `ArrayDefinition` or `ArraySpecification` or something. This isn't actually an array but a statement that defines a schema which validates against an array.